app/services/task_service/gpt_generate_tests.rb

```diff
@@ -0,0 +1,83 @@
+# frozen_string_literal: true
+
+module TaskService
+  class GptGenerateTests < ServiceBase
+    def initialize(task:)
+      super()
+      @task = task
+      @message = task.description
+      if @task.programming_language&.language
+        @language = @task.programming_language.language
+      else
+        raise MissingLanguageError
+      end
```

Comment:
We can probably simplify this:

```suggestion
      @language = @task.programming_language&.language
      raise MissingLanguageError if @language.blank?
```

Reply:
I also switched the notice here to alert.